product/views.py

```diff
@@ -0,0 +1,184 @@
+
+from django.views import View
+from django.http  import JsonResponse
+from .models      import MainCategory, SubCategory, Product
+import re
+
+class MainCategoryView(View):   # 메인 카테고리 리스트 뷰 / 토이 or 문구
+    def get(self, request, main_categories_id):
+        
+        if not Product.objects.filter(main_category_id=main_categories_id).exists():
+            return JsonResponse({'message': 'NOT FOUND'}, status=404)
+
+        main_category_data = Product.objects.filter(main_category_id=main_categories_id)
```

Comment:
Product.object.filter()를 하는 경우 product QuerySet[]으로 반환합니다. 그렇기 때문에, 아래와 같이 수정하는게 맞습니다.
```suggestion
        products = Product.objects.filter(main_category_id=main_categories_id)
```